ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -562,57 +629,500 @@ toPublicState PeerSelectionState { knownPeers
+      numberOfColdBootstrapPeersPromotions,
+      numberOfEstablishedBootstrapPeers          = numberOfEstablishedBootstrapPeers
+                                                 - numberOfActiveBootstrapPeers,
+      numberOfWarmBootstrapPeersDemotions,
+      numberOfWarmBootstrapPeersPromotions,
+      numberOfActiveBootstrapPeers,
+      numberOfActiveBootstrapPeersDemotions
+    }
+
+
+-- | Compute peer selection sets & their sizes.
+--
+-- This function is used internally by the outbound-governor and to compute
+-- `PeerSelectionCounters` which are used by churn or are traced (e.g. as EKG
+-- metrics).  For this reason one has to be very careful when changing the
+-- function, as it will affect the outbound governor behaviour.
+--
+peerSelectionStateToView
+  :: Ord peeraddr
+  => PeerSelectionState peeraddr peerconn
+  -> PeerSelectionSetsWithSizes peeraddr
+peerSelectionStateToView
+    PeerSelectionState {
+        knownPeers,
+        establishedPeers,
+        activePeers,
+        publicRootPeers,
+        localRootPeers,
+        inProgressPromoteCold,
+        inProgressPromoteWarm,
+        inProgressDemoteWarm,
+        inProgressDemoteHot
+      }
+    =
+    PeerSelectionView {
+      viewRootPeers                          = size rootPeersSet,
+
+      viewKnownPeers                         = size   knownPeersSet,
+      viewAvailableToConnectPeers            = size $ availableToConnectSet
+                                                      Set.\\ bigLedgerSet,
+      viewColdPeersPromotions                = size $ inProgressPromoteCold
+                                                      Set.\\ bigLedgerSet,
+      viewEstablishedPeers                   = size   establishedPeersSet,
+      viewWarmPeersDemotions                 = size $ inProgressDemoteWarm
+                                                      Set.\\ bigLedgerSet,
+      viewWarmPeersPromotions                = size $ inProgressPromoteWarm
+                                                      Set.\\ bigLedgerSet,
+      viewActivePeers                        = size $ activePeersSet,
+      viewActivePeersDemotions               = size $ activePeersSet
+                                                      `Set.intersection` inProgressDemoteHot,
+
+      viewKnownBigLedgerPeers                = size   bigLedgerSet,
+      viewAvailableToConnectBigLedgerPeers   = size $ availableToConnectSet
+                                                      `Set.intersection` bigLedgerSet,
+      viewColdBigLedgerPeersPromotions       = size $ bigLedgerSet
+                                                      `Set.intersection` inProgressPromoteCold,
+      viewEstablishedBigLedgerPeers          = size   establishedBigLedgerPeersSet,
+      viewWarmBigLedgerPeersDemotions        = size $ inProgressDemoteWarm
+                                                      `Set.intersection` bigLedgerSet,
```

Comment:
bigLedgerSet -> establishedBigLedgerPeerSet since it is smaller

Reply:
Addressed.